### DISCUSSION TITLE
Language switch keybinding stops working on external usb keyboard post sleep

### DISCUSSION THREAD

**Discussion by shaief** [i3/i3#5350] [Q&A]
My setup uses a pretty standard keybinding for language switch (alt+shift):
```
exec --no-startup-id "setxkbmap -layout us,il -option grp:alt_shift_toggle -option 'grp_led:caps'"
```
It works well on my laptop keyboard, but when i'm using an external keyboard it works only until i close the lid of my laptop and put it to sleep. Post waking up - the keybinding stops working on my usb/external keyboard but keeps working on my laptop keyboard.

What am i missing? What should i do to make it work on all keyboards all the time?

Thanks!

**Comment by neg-serg** [comment]
Try some alternative options for example:
1) /etc/default/keyboard or ~/.dmrc. I've not tried it but it seems to work for some people. Then you need to make `sudo udevadm trigger --subsystem-match=input --action=change` and check that settings are working. You can experiment with it.
2) localectl can also save settings permamently to Xorg config. Then you need go reboot.
